docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+## Examples
+
+Simplest CLI command for a binary classification problem (AutoML will need to infer most of the configuration from the provided data):
+
+(*Release 0.1 examples*)
+
+` mlnet new --ml-task BinaryClassification --dataset "/MyDataSets/Cars.csv"`
+
+Another simple CLI command for a regression problem where the target-column is provided explecitely:
+
+` mlnet new --ml-task Regression --dataset "/MyDataSets/Sales.csv" --label-column "MonthlySales" `
+
+Create and train a binary-classification model with quite a few explicit arguments:
+
+` mlnet new --ml-task BinaryClassification --app-type console --dataset "/MyDataSets/Population-Training.csv" --test-dataset "/MyDataSets/Population-Test.csv" --label-column-name "InsuranceRisk" --max-exploration-time 600`
+
+Create and train a model based on arguments specified in the CLI response file (.rsp) with arguments containing the configuration:
+
+` mlnet new @my_cli_config_args.rsp`
+
+
+(*Release 0.2 examples*)
+
+Simplest command where the tool infers the type of ML taks to perform based on the data:
+
+` mlnet new --dataset "/MyDataSets/Cars.csv"`
+
+Create and train a model based on parameters specified in the .rsp file plus more advanced model settings in the .yaml file:
+
+` mlnet new @my_cli_config_args.rsp --model-settings-file "./my_model_settings.yaml"  `
+
+
+(*Release v1.0 examples*)
+
+CLI command specifying to run AutoML compute in Azure's cloud 
+
+` mlnet new --ml-task BinaryClassification --dataset "/MyDataSets/Cars.csv" --label-column-name "InsuranceRisk" --azure-automl-service-key "key-value"`
+
+
+
+Syntax should be aligned to [dotnet new](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new?tabs=netcore21#synopsis).
+
+#### Syntax
+
+```console
+mlnet new 
+
+--------------- (v0.1) -------------------
+
+--ml-task <value>
+
+--dataset <value>
+[
+ [--validation-dataset <value>]
+  --test-dataset <value>
+]
+
+--label-column-name <value>
+|
+--label-column-index <value>
+
+[--has-header <value>]
+
+[--max-exploration-time <value>]
+[--verbosity <value>]
+
+[--name <value>]
+[--list-ml-tasks]
+[-h|--help]
+
+-------------- (v0.2) --------------------
+[--ml-task <value>]
+[--model-settings-file <value>]
+[<RSP FILE NAME>]
+[--best-models-count]
+[--app-type <value>]
+[--output-folder <value>]
+[--force]
+
+-------------- (v1.0) --------------------
+[--list-app-types]
+[--debug-log-file <value>]
+------------------------------------------
+
+-------------- (vNext) -------------------
+[-lang|--language <value>]
+------------------------------------------
+
+```
+
+#### Description
+
+The `mlnet new` command provides a CLI oriented way to create projects or solutions such as:
+
+- Create a single project (console app) with:
+    - *Training ML.NET code:* One seggregated method per ranked model, but part of the same console app.
+    - *Scoring/consuming ML.NET code:* One seggregated method per ranked model, but part of the same console app.
```

Comment:
> One seggregated method per ranked model [](start = 38, length = 40)

Can we get a way with a single method?  All the signatures of all models will be the same. Only thing different is which zip file to load.